src/replication.c

```diff
@@ -1716,12 +1718,13 @@ int istgt_execute_volume_operation(spec_t *spec, zvol_op_code_t opcode,
 	io_seq = ++spec->io_seq;
 	cf = spec->consistency_factor;
 	rf = spec->replication_factor;
+	ret = 0;
 
 	rcommon_mgmt_cmd_t *rmgmt = allocate_rcommon_mgmt_cmd(0);
 	replica_t *hr = spec->rebuild_info.healthy_replica;
 	if (hr && opcode == ZVOL_OPCODE_SNAP_CREATE) {
-		BUILD_VOL_CMD_DATA(spec)
 		REPLICA_LOG("sending SNAP_PREP to Replica(%lu)\n", hr->zvol_guid);
+		BUILD_VOL_CMD_DATA(spec, opcode, snapname, size, data, data_len);
```

Comment:
This macro is returning a value in case of invalid opcode, without unlocking a `spec->rq_mtx`!

Reply:
I will update it.